src/textacy/extract/triples.py

```diff
@@ -317,3 +318,64 @@ def expand_verb(tok: Token) -> list[Token]:
         child for child in tok.children if child.dep in _VERB_MODIFIER_DEPS
     ]
     return [tok] + verb_modifiers
+
+def filter_cue_candidates(tok, verbs):
+    return all([
+        tok.pos == VERB,
+        tok.lemma_ in verbs
+    ])
+
+# def get_cue_candidates(window_sents, verbs):
+#     return [
+#             tok
+#             for sent in window_sents
+#             for tok in sent
+#             if filter_cue_candidates(tok, verbs)
+#         ]
+
+def filter_speaker_candidates(candidate, i, j):
+    """
+    Evaluates whether the candidate is not punctuation and that it is outside of the quote. 
+    """
+    return all([
+            candidate.pos!=PUNCT,
+            ((candidate.i >= i and candidate.i >= j) or (candidate.i <= i and candidate.i <= j)),
+        ])
+
+def line_break_window(i, j, doc):
+    """
+    Finds the boundaries of the paragraph containing doc[i:j].
+    """
+    for i_, j_ in list(zip(
+        [tok.i for tok in doc if tok.text=="\n"],
+        [tok.i for tok in doc if tok.text=="\n"][1:])
+        ):
```

Comment:
trying to improve efficiency/performance here

```suggestion
    lb_tok_idxs = [tok.i for tok in doc if tok.text == "\n"]
    for i_, j_ in zip(lb_tok_idxs, lb_tok_idxs[1:]):
```